clients/include/testing_axpy_batched_ex.hpp

```diff
@@ -40,21 +36,18 @@ hipblasStatus_t testing_axpy_batched_ex_template(const Arguments& argus)
     hipblasDatatype_t yType         = argus.c_type;
     hipblasDatatype_t executionType = argus.compute_type;
 
-    int abs_incx = incx < 0 ? -incx : incx;
     int abs_incy = incy < 0 ? -incy : incy;
 
-    size_t sizeX   = size_t(N) * abs_incx;
-    size_t sizeY   = size_t(N) * abs_incy;
-    Ta     h_alpha = argus.get_alpha<Ta>();
+    Ta h_alpha = argus.get_alpha<Ta>();
 
     // Naming: dX is in GPU (device) memory. hK is in CPU (host) memory, plz follow this practice
-    host_batch_vector<Tx> hx(N, incx, batch_count);
-    host_batch_vector<Ty> hy_host(N, incy, batch_count);
-    host_batch_vector<Ty> hy_device(N, incy, batch_count);
-    host_batch_vector<Ty> hy_cpu(N, incy, batch_count);
+    host_batch_vector<Tx> hx(N, incx ? incx : 1, batch_count);
+    host_batch_vector<Ty> hy_host(N, incy ? incy : 1, batch_count);
+    host_batch_vector<Ty> hy_device(N, incy ? incy : 1, batch_count);
+    host_batch_vector<Ty> hy_cpu(N, incy ? incy : 1, batch_count);
```

Comment:
Good point, I'll do that.
We don't mention anything in our API about inc == 0 for functions that support inc < 0. If it's writing to the vector with the 0 increment I would imagine it would be a race condition. If it's just reading it would just read the first element over and over - I don't see anything mentioned in the lapack [saxpy documentation](http://www.netlib.org/lapack/explore-html/df/d28/group__single__blas__level1_gad2a52de0e32a6fc111931ece9b39726c.html#gad2a52de0e32a6fc111931ece9b39726c)

Reply:
Well if you want to match cublas then you'll need to pass the 0 cases through and not just return success.  The allocs can also just allocate 1 entry if increment is 0.   Fine by me.  Maybe Andrew has an opinion on this?

Reply:
Made a change for this, I haven't checked if it all passes on cuBLAS yet.